libagent/device/ledger.py

```diff
@@ -72,7 +72,7 @@ def pubkey(self, identity, ecdh=False):
     def sign(self, identity, blob):
         """Sign given blob and return the signature (as bytes)."""
         path = _expand_path(identity.get_bip32_address(ecdh=False))
-        if identity.identity_dict['proto'] == 'ssh':
+        if identity.identity_dict['proto'] == 'ssh' and len(blob) != 32:
```

Comment:
this line is somewhat mysterious to me

Reply:
I guess that what's happening is, in our case the proto is ssh and the blob is a 32 byte merlin transcript.

So we are changing from selecting the ssh codepath to a non-ssh codepath, and this is changing the ins byte in the protocol.